8-application-demos/6-kalshi-bet-predictor/compare.py

```diff
@@ -0,0 +1,127 @@
+import csv
+import json
+from typing import Dict, List, Tuple
+import asyncio
+import aiohttp
+
+def load_markets(csv_path: str) -> List[Tuple[str, str]]:
+    markets = []
+    with open(csv_path, 'r') as f:
+        reader = csv.reader(f)
+        next(reader) # skip header
+        for row in reader:
+            if len(row) >= 2:
+                markets.append((row[0], row[1]))
+    return markets
+
+async def get_market_data(session: aiohttp.ClientSession, kalshi_ticker: str, 
+                         polymarket_slug: str, endpoint_url: str) -> Dict:
+    
+    payload = json.dumps({
+        'kalshi_ticker': kalshi_ticker,
+        'poly_slug': polymarket_slug
+    })
+    
+    headers = {
+        'Authorization': '<YOUR AUTHORIZATION>',
+        'Content-Type': 'application/json'
+    }
+    
+    try:
+        async with session.post(endpoint_url, headers=headers, data=payload) as response:
+            response.raise_for_status()
+            data = await response.json()
+            print(data)
+            data = data['result']
+            
+            kalshi_data = data['kalshi']
+            polymarket_data = data['polymarket']
+            
+            return {
+                'kalshi_ticker': kalshi_ticker,
+                'polymarket_slug': polymarket_slug,
+                'kalshi_edge': kalshi_data['edge'],
+                'polymarket_edge': polymarket_data['edge'],
+                'kalshi_buy_yes': kalshi_data['buy_yes'],
+                'kalshi_buy_no': kalshi_data['buy_no'],
+                'polymarket_buy_yes': polymarket_data['buy_yes'],
+                'polymarket_buy_no': polymarket_data['buy_no'],
+            }
+    except Exception as e:
+        print(f"Error fetching data for {kalshi_ticker}/{polymarket_slug}: {e}")
+        return None
+
+async def analyze_markets_async(csv_path: str, endpoint_url: str) -> List[Dict]:
+    markets = load_markets(csv_path)
+    
+    print(f"Fetching data for {len(markets)} markets all at once...")
+    
+    async with aiohttp.ClientSession() as session:
+        tasks = [get_market_data(session, kalshi_ticker, polymarket_slug, endpoint_url) 
```

Comment:
strange its kalshi_ticket and polymarket_slug. You can keep this on your cerebrium api but should have consistency here